src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -5067,8 +5066,9 @@ impl<'a> TypeChecker<'a> {
     fn try_fold_constant<Index: ColumnIndex>(
         &self,
         expr: &EExpr<Index>,
+        enable_shrink: bool,
     ) -> Option<Box<(ScalarExpr, DataType)>> {
-        if expr.is_deterministic(&BUILTIN_FUNCTIONS) {
+        if expr.is_deterministic(&BUILTIN_FUNCTIONS) && enable_shrink {
```

Comment:
we should always do constant folder.

enable_shrink need to be checked inside

Reply:
Done